packages/schematics/angular/migrations/update-6/index.ts

```diff
@@ -0,0 +1,267 @@
+  };
+}
+
+// tslint:disable-next-line:no-any
+function migrateConfiguration(oldConfig: any): Rule {
+  const oldConfigPath = '.angular-cli.json';
+  const configPath = 'angular.json';
+
+  return (host: Tree, context: SchematicContext) => {
+    context.logger.info(`Updating configuration`);
+    // tslint:disable-next-line:no-any
+    const config: any = {
+      version: 1,
+      newProjectRoot: 'projects',
+      cli: extractCliConfig(oldConfig),
+      schematics: extractSchematicsConfig(oldConfig),
+      architect: extractArchitectConfig(oldConfig),
+      projects: extractProjectsConfig(oldConfig),
+    };
+
+    context.logger.info(`Removing old config file (${oldConfigPath})`);
+    host.delete(oldConfigPath);
+    context.logger.info(`Writing config file (${configPath})`);
+    host.create(configPath, JSON.stringify(config, null, 2));
+
+    return host;
+  };
+}
+
+// tslint:disable-next-line:no-any
+function extractCliConfig(_config: any): JsonObject {
+  // tslint:disable-next-line:no-any
+  const cfg: any = {};
+
+  return cfg;
+}
+
+// tslint:disable-next-line:no-any
+function extractSchematicsConfig(config: any): JsonObject {
+  let collectionName = '@schematics/angular';
+  if (config.schematics && config.schematics.collection) {
+    collectionName = config.schematics.collection;
+  }
+
+  /**
+   * For each schematic
+   *  - get the config
+   *  - filter one's without config
+   *  - combine them into an object
+   */
+  // tslint:disable-next-line:no-any
+  const schematicConfigs: any = ['class', 'component', 'directive', 'guard', 'interface', 'module',
+                                 'pipe', 'service']
+    .map(schematicName => {
+      return {
+        schematicName,
+        config: config.defaults[schematicName] || null ,
+      };
+    })
+    .filter(schematic => schematic.config !== null)
+    .reduce((all: JsonObject, schematic) => {
+      all[collectionName + ':' + schematic.schematicName] = schematic.config;
+
+      return all;
+    }, {});
+
+  const componentKey = collectionName + ':component';
+  const directiveKey = collectionName + ':directive';
+  if (!schematicConfigs[componentKey]) {
+    schematicConfigs[componentKey] = {};
+  }
+  if (!schematicConfigs[directiveKey]) {
+    schematicConfigs[directiveKey] = {};
+  }
+  schematicConfigs[componentKey].prefix = config.apps[0].prefix;
+  schematicConfigs[directiveKey].prefix = config.apps[0].prefix;
+  schematicConfigs[componentKey].styleext = config.defaults.styleExt;
+
+  return schematicConfigs;
+}
+
+// tslint:disable-next-line:no-any
+function extractArchitectConfig(_config: any): JsonObject {
+  return {};
+}
+// tslint:disable-next-line:no-any
+function extractProjectsConfig(config: any): any {
+  const builderPackage = '@angular-devkit/build-webpack';
+  // convert the apps to projects
+  // tslint:disable-next-line:no-any
+  const projectMap = config.apps.map((app: any, idx: number) => {
+    const name = app.name || `$$proj${idx}`;
```

Comment:
Maybe we should review this name for the first app, which is the one most people have. So

```
const name = app.name || idx === 0 ? 'app' : `$$proj${idx}`;
```

WDYT?